WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerSearchView.java

```diff
@@ -64,9 +63,7 @@ public boolean onMenuItemActionCollapse(MenuItem item) {
     public void enableSearchMode() {
         mMenuEdit.setVisible(false);
         mSitePickerActivity.setIsInSearchModeAndNullifyAdapter(true);
-        SitePickerSearchAdapter adapter = (SitePickerSearchAdapter) mSitePickerActivity.getAdapter();
-        mSitePickerActivity.getRecycleView().swapAdapter(adapter, true);
-        adapter.loadSites();
+        mSitePickerActivity.getRecycleView().swapAdapter(mSitePickerActivity.getAdapter(), true);
```

Comment:
> Is there a way `SitePickerSearchView` can be written so it doesn't hold a reference to `SitePickerActivity`?

`SitePickerSearchView` was written so that all work related to search was contained in one place. That means it needs a reference to the Activity to do some work. Do you have any ideas on how I could improve this while keeping the logic out of the Activity? I'm all for removing a dependency.

> right now both `SitePickerActivity` and `SitePickerAdapter` have their own mIsInSearchMode and mLastSearch variables, which seems error prone. I think these can be handled solely by the adapter.

Unfortunately on rotation, the adapter gets nulled out. I needed to keep track of whether the user was in search mode so I could initialize the adapter in the correct way. Is there a way you know of that could work? Only way I could think of is preferences, but that's not desirable IMO.